File: HackerRank 30 Days/day4/main.py
class Person:
    def __init__(self, initialAge: int):
        if initialAge < 0:
            print("Age is not valid, setting age to 0.")
            self.age = 0
        else:
            self.age = initialAge

    def amIOld(self) -> None:
        if self.age < 13:
            print("You are young.")
        elif self.age < 18:
            print("You are a teenager.")
        else:
            print("You are old.")

    def yearPasses(self) -> None:
        self.age += 1


def main() -> None:
    t = int(input().strip())

    for _ in range(t):
        age = int(input().strip())
        person = Person(age)
        person.amIOld()
        for _ in range(3):
            person.yearPasses()
        person.amIOld()


if __name__ == "__main__":
    main()

File: HackerRank 30 Days/day4/main.js
'use strict';

class Person {
    constructor(initialAge) {
        if (initialAge < 0) {
            this.age = 0;
            console.log("Age is not valid, setting age to 0.");
        } else {
            this.age = initialAge;
        }
    }

    amIOld() {
        if (this.age < 13) {
            console.log("You are young.");
        } else if (this.age >= 13 && this.age < 18) {
            console.log("You are a teenager.");
        } else {
            console.log("You are old.");
        }
    }

    yearPasses() {
        this.age += 1;
    }
}


// Código principal
const fs = require('fs');

const input = fs.readFileSync(0, 'utf8').trim().split(/\s+/).map(Number);

let index = 0;
const T = input[index++];

for (let i = 0; i < T; i++) {
    const age = input[index++];

    const p = new Person(age);

    p.amIOld();

    for (let j = 0; j < 3; j++) {
        p.yearPasses();
    }

    p.amIOld();

    console.log();
}